src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -69,14 +69,11 @@ impl Config {
                 .coords_check(meta, q_mul_fixed_running_sum, word)
         });
 
-        // Check that we get z_85 = 0 as the final output of the three-bit decomposition running sum.
-        // Also check that the base field element is canonical.
+        // Check that the base field element is canonical.
```

Comment:
https://zcash.github.io/orchard/design/circuit/gadgets/ecc/fixed-base-scalar-mul.html#fixed-base-scalar-multiplication-using-base-field-element says that z_85 must be constrained to 0. Do we already constrain it elsewhere? If we do, then add a comment saying where.

Reply:
`z_85 = 0` is now constrained by the `self.running_sum_config.copy_decompose()` call in `base_field_elem::Config_assign()`, with `strict = true`.

I'll add a comment to this effect + update the book.

Reply:
The comment should go on `Config::create_gate`, which is where it is relevant.

@daira `z_85` is constrained because we now use strict checks during scalar decomposition.

Reply:
I don't see any uses of `strict = false` except in tests. Did I miss any? If not, why do we need the strict check to be optional?

Reply:
The updated book PR is at: https://github.com/zcash/orchard/pull/146.

Reply:
I chose to include the `strict` option for a more general helper API.

(Sidenote: For the `lookup_range_check` helper, we only ever use `strict = false`, so we could choose to not provide a `strict = true` option there.)

Reply:
I find the code more confusing to think about with the two options, and more work to review. Let's remove the ones we don't use in Orchard.